go_/models/board.go

```diff
@@ -0,0 +1,104 @@
+package models
+
+import (
+    "errors"
+    "fmt"
+    "strings"
+)
+
+var defaultCell = "·"
+
+type Board struct {
+    rows [][]string
+}
+
+func (b *Board) Print() {
+    fmt.Println("")
+
+    for i := 0; i < len(b.rows); i++ {
+        fmt.Printf("  | %s |\n", strings.Join(b.rows[i], " "))
+    }
+    fmt.Printf("  *%s*\n", strings.Repeat("‾", len(b.rows[0]) * 2 + 1))
+    fmt.Print("   ")
+
+    for i, _ := range b.rows[0] {
+        fmt.Printf(" %v", i + 1)
+    }
+    fmt.Println("\n")
+}
+
+func (b *Board) Receive(col int, token string) (int, error) {
+    if col < 0 || col > 6 {
+        return -1, errors.New("Column is out of range")
+    }
+
+    for row := 6; row >= 0; row-- {
+        if b.rows[row][col] == defaultCell {
+            b.rows[row][col] = token
+            return row, nil
+        }
+    }
+    return -1, errors.New("Column is full")
+}
+
+func (b *Board) Inspect(row, col int) (win bool, draw bool) {
+    win =
+        b.wonVertical(row, col) || b.wonHorizontal(row, col) ||
+        b.wonDiagonal(row, col) || b.wonAntidiagonal(row, col)
```

Comment:
I bet there's some crazy way to bit-shift the checking for wins, but I'm not smart

Reply:
Computers are fast so we don't have to be